File: python/sorting/insertion_sort.py
from .sort import SortType, Sort


class InsertionSort(Sort):
    def sort(self, arr: list[int], type: SortType = SortType.asc):
        length = len(arr)

        for i in range(1, length):
            j = i
            while j >= 1 and super().compare(arr[j - 1], arr[j], type):
                arr[j - 1], arr[j] = arr[j], arr[j - 1]
                j += -1

        return arr

File: typescript/sorting/insertion_sort.ts
import Sort, { SortType, SORT_TYPE } from "./sort.ts";

class InsertionSort extends Sort {
  sort(arr: number[], type: SortType = SORT_TYPE.asc): number[] {
    const length: number = arr.length;

    for (let i = 1; i < length; i++) {
      let j = i;
      while (j >= 1 && super.compare(arr[j - 1], arr[j], type)) {
        [arr[j - 1], arr[j]] = [arr[j], arr[j - 1]];
        j--;
      }
    }
    return arr;
  }
}

export default InsertionSort;
